src/hotspot/share/opto/graphKit.cpp

```diff
@@ -1565,6 +1566,9 @@ Node* GraphKit::store_to_memory(Node* ctl, Node* adr, Node *val, BasicType bt,
   if (unsafe) {
     st->as_Store()->set_unsafe_access();
   }
+  if (barrier_data != 0) {
```

Comment:
Do we need this check? The analogous load construction logic (https://github.com/openjdk/jdk/blob/17a3f0e2577f2f9eb3fe62a4b8261e3dbe4c3b28/src/hotspot/share/opto/memnode.cpp#L951) does not have it.

Reply:
I'll remove the check for consistency.